StardewArchipelago/GameModifications/Buildings/CarpenterMenuArchipelago.cs

```diff
@@ -19,24 +20,28 @@ public CarpenterMenuArchipelago(IModHelper modHelper, ArchipelagoClient archipel
         public override List<BluePrint> GetAvailableBlueprints()
         {
             var blueprints = new List<BluePrint>();
-
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_COOP);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_BARN);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_WELL);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_SILO);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_MILL);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_SHED);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_FISH_POND);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_STABLE, true);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_SLIME_HUTCH);
-
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_BIG_COOP, requiredBuilding: CarpenterInjections.BUILDING_COOP);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_DELUXE_COOP, requiredBuilding: CarpenterInjections.BUILDING_BIG_COOP);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_BIG_BARN, requiredBuilding: CarpenterInjections.BUILDING_BARN);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_DELUXE_BARN, requiredBuilding: CarpenterInjections.BUILDING_BIG_BARN);
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_BIG_SHED, requiredBuilding: CarpenterInjections.BUILDING_SHED);
-
-            AddBuildingBlueprintIfReceived(blueprints, CarpenterInjections.BUILDING_SHIPPING_BIN);
+            var blueprintData = FullBlueprintData();
+            foreach (var blueprint in blueprintData)
+            {
+                var blueprintMagical = blueprint.Value.magical;
+                var blueprintUpgrade = blueprint.Value.blueprintType;
+
+                if (blueprintMagical)

```

Comment:
This is also now the exact opposite of what it was before, therefore wrong

Reply:
Went with the invert if idea.